### DISCUSSION TITLE
How to resolve this build error in typescript?

### DISCUSSION THREAD

**Discussion by Hermestech** [microsoft/fluentui#24614] [Q&A] [answered]
I am implementing fluentui in nextjs according to the documentation on this page: https://react.fluentui.dev/?path=/docs/concepts-developer-server-side-rendering--page.

If i´m build with the extension of `app.jsx` it´s ok, any problem. But when i write the extension `_app.tsx ` I have a problem when I try to build the application, because it returns the following errors: _**Type '{ children: Element; renderer: any; }' is not assignable to type 'IntrinsicAttributes & RendererProviderProps'. Property 'children' does not exist on type 'IntrinsicAttributes & RendererProviderProps'**_

```
`import {
  createDOMRenderer,
  RendererProvider,
  SSRProvider,
  FluentProvider,
  webLightTheme
} from '@fluentui/react-components'
import '../styles/styles.css'


function MyApp({ Component, pageProps, renderer }) {
  return (
      <RendererProvider renderer={renderer || createDOMRenderer()}>
        <SSRProvider>
          <FluentProvider theme={webLightTheme}>
            <Component {...pageProps} />
          </FluentProvider>
        </SSRProvider>
      </RendererProvider>
  )
}

export default MyApp`

```

i leave a screenshot from the error: 
![Captura de pantalla 2022-08-31 114714](https://user-images.githubusercontent.com/64867364/187734208-73404042-a3f2-4f0d-856f-db5293f6037c.png)

**Comment by micahgodbolt** [comment, marked as answer]
We are still working on support for React 18. Try downgrading to React 17.

**Reply by layershifter** [reply]
Indeed, Griffel i.e. `RendererProvider` component does not support `@types/react@18`, https://github.com/microsoft/griffel/issues/40.
